src/Generating/MineShafts.cpp

```diff
@@ -767,27 +767,34 @@ void cMineShaftCorridor::PlaceChest(cChunkDesc & a_ChunkDesc)
 	int BlockZ = a_ChunkDesc.GetChunkZ() * cChunkDef::Width;
 	int x, z;
 	NIBBLETYPE Meta = 0;
-	switch (m_Direction)
-	{
-		case dirXM:
-		case dirXP:
+	std::tie(x, z, Meta) = [&]() -> std::tuple<int, int, NIBBLETYPE>
 		{
-			x = m_BoundingBox.p1.x + m_ChestPosition - BlockX;
-			z = m_BoundingBox.p1.z - BlockZ;
-			Meta = E_META_CHEST_FACING_ZP;
-			break;
-		}
+			switch (m_Direction)
+			{
+				case dirXM:
+				case dirXP:
+				{
+					return
+					{
+						m_BoundingBox.p1.x + m_ChestPosition - BlockX,
```

Comment:
You could also modify the variables directly from the lambda, like it was in the old code (just capture them by reference). This could look a bit better than the tuples.